services/src/main/java/org/keycloak/cookie/SecureContextResolver.java

```diff
@@ -0,0 +1,32 @@
+package org.keycloak.cookie;
+
+import java.net.URI;
+
+class SecureContextResolver {
+
+    static boolean isSecureContext(URI uri) {
```

Comment:
This method does not check for an exact match on `localhost.` or hosts ending in `.localhost.` (see [specification](https://w3c.github.io/webappsec-secure-contexts/#algorithms)). For example the following URLs are considered secure:

```
http://localhost.
http://foo.localhost.
```

Reply:
Added, somehow missed those. Never seen anyone use `localhost.` ;)

Reply:
Yeah don't see anyone using this, but surely there is one user out there that will complain :P